make/photon/db/docker-entrypoint.sh

```diff
@@ -1,120 +1,59 @@
-                        WARNING: No password has been set for the database.
-                                        This will allow anyone with access to the
-                                        Postgres port to access your database. In
-                                        Docker's default configuration, this is
-                                        effectively any other container on the same
-                                        system.
-                                        Use "-e POSTGRES_PASSWORD=password" to set
-                                        it in "docker run".
-                        ****************************************************
-EOF
-
-                pass=
-                authMethod=trust
+source $PWD/initdb.sh
+
+CUR=$PWD
+TO_UPGRADE=false
+PG_VERSION_NEW=$1
+
+PGDATAOLD=""
+PGDATANEW=${PGDATA}/pg${PG_VERSION_NEW}
+
+#
+# Init DB: $PGDATA is empty.
+# Upgrade DB: 1, has $PGDATA\PG_VERSION. 2, has pg version directory with PG_VERSION inside.
+# Error: $PGDATA is not empty and has files or invalid directory.
+#
+if [ "$(ls -A $PGDATA)" ]; then
+        if [ ! -d $PGDATANEW ]; then
+                TO_UPGRADE=true
+                if [ -s $PGDATA/PG_VERSION ]; then
+                        PGDATAOLD=$PGDATA
+                else
+                        # see the first directory as the PGDATAOLD
+                        files=($PGDATA/*)
+                        oldpg=$(echo "${files[0]}")
+                        PGDATAOLD=$PGDATA/oldpg
+                        if [ ! -d $PGDATAOLD ] || [ ! -s $PGDATAOLD/PG_VERSION ]; then
+                                echo "incorrect data: $PGDATA, make sure $PGDATA is empty or only contains one pg version directory with PG_VERSION inside."
+                                exit 1
+                        fi
+                fi
         fi
+else
+        initPG $PGDATANEW true
+fi
 
-        {
-                echo
-                echo "host all all all $authMethod"
-        } >> "$PGDATA/pg_hba.conf"
-        echo `whoami`
-        # internal start of server in order to allow set-up using psql-client
-        # does not listen on external TCP/IP and waits until start finishes
-        pg_ctl -D "$PGDATA" -o "-c listen_addresses=''" -w start
-
-        file_env 'POSTGRES_USER' 'postgres'
-        file_env 'POSTGRES_DB' "$POSTGRES_USER"
-
-        psql=( psql -v ON_ERROR_STOP=1 )
-
-        if [ "$POSTGRES_DB" != 'postgres' ]; then
-                "${psql[@]}" --username postgres <<-EOSQL
-                        CREATE DATABASE "$POSTGRES_DB" ;
-EOSQL
-                echo
+if [ $TO_UPGRADE == true ]; then
+        initPG $PGDATANEW false
+        bash "$CUR/upgrade.sh" --old-datadir $PGDATAOLD --new-datadir $PGDATANEW
+        if [ $? -ne 0 ]; then
+                exit 1
         fi
-
-        if [ "$POSTGRES_USER" = 'postgres' ]; then
-                op='ALTER'
+        if [ "$PGDATAOLD" = "$PGDATA" ]; then
```

Comment:
print a log for removing the `$PGDATAOLD`?

Reply:
added